src/components/widget/List/RawList.js

```diff
@@ -146,46 +174,61 @@ class RawList extends PureComponent {
     }
 
     const { top } = this.list.getBoundingClientRect();
-    const { filter, isToggled, onCloseDropdown } = this.props;
+    const { handleCloseDropdown } = this;
+    const { filter, isToggled } = this.props;
+
     if (
       isToggled &&
       filter.visible &&
       (top + 20 > filter.boundingRect.bottom ||
         top - 20 < filter.boundingRect.top)
     ) {
-      onCloseDropdown();
+      handleCloseDropdown();
     }
   };
 
-  handleClickOutside() {
-    const { isToggled, onCloseDropdown, onBlur, selected } = this.props;
+  handleClickOutside = event => {
+    const { list, handleCloseDropdown } = this;
+    const { isToggled } = this.props;
+
+    if (!list || !isToggled) {
+      return;
+    }
+
+    const { top, right, bottom, left } = list.getBoundingClientRect();
+
+    if (
```

Comment:
I was thinking maybe we can use this instead : https://github.com/Pomax/react-onclickoutside#marking-elements-as-skip-over-this-one-during-the-event-loop ? 

But I'm not 100% sure we even need this.